docs/reference/src/language/builtins/elements.md

```diff
@@ -253,18 +253,24 @@ export component Example inherits Window {
 
 ## `GridLayout`
 
-`GridLayout` places its children in a grid. `GridLayout` adds properties to each child: `col`, `row`, `colspan`, `rowspan`.
-These properties must be compile time constants.
-You can control the position of children with `col` and `row`.
-If `col` or `row` aren't specified, they are automatically computed such that the item is next to the previous item, in the same row.
-Alternatively, the item can be put in a `Row` element.
+`GridLayout` places elements on a grid.
+
+It adds the following properties to each cell element: `col`, `row`, `colspan`, `rowspan`.
+These properties must be compile-time constants.
+
+`col` and `row` control the position of a cell on the grid structure.
+`row` must be specified explicitly to be able to advance it&mdash;or it'll stay the same&mdash;which then implicitly resets an unspecified `col` to `0`.
+Wrapping elements in `Row` elements instead will take care of row advancement for you.
+Omitting `col` implicitly means the next column (or the first at the beginning).
+
+If the cell elements' size constraints don't work against this, a `GridLayout` covers its whole surface with its cells. The elements constituting the cells will be stretched inside their allotted space.
```

Comment:
It's something you have to learn about `GridLayout` (and something that maybe should change by supporting alignment on both axes), because other layouts like `VerticalLayout` allow you to align the whole block of elements at one side. The very last sentence with "allotted space" is the part that made me edit this section of the docs to add it.

In my app, e.g., I have form widgets in a `GridLayout`, so the text boxes on the right are nicely aligned with each other, and need to wrap it in a `VerticalLayout` with `alignment: start;` to be able to present the form properly and not stretched.

Reply:
Thanks for the explanation. Mentioning alignment etc. makes sense.